monetio/sat/_tempo_l2_no2_mm.py

```diff
@@ -0,0 +1,146 @@
+"""TEMPO L2 NO2 data file reader.
+
+History:
+
+
+"""
+
+import logging
+import os
+import sys
+from collections import OrderedDict
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from netCDF4 import Dataset
+
+
+def _open_one_dataset(fname, variable_dict):
+    """Read locally stored INTELSAT TEMPO NO2 level 2
+    Parameters
+    ----------
+    fname: string
+        fname is local path to netcdf4 (HDF5) file
+
+    variable_dict: dict
+    Returns
+    -------
+    ds: xarray dataset
+    """
+    print("reading " + fname)
+
+    ds = xr.Dataset()
+
+    dso = Dataset(fname, "r")
+    lon_var = dso.groups["geolocation"]["longitude"]
+    lat_var = dso.groups["geolocation"]["latitude"]
+    time_var = dso.groups["geolocation"]["time"]
+
+    ds["lon"] = (
+        ("x", "y"),
+        lon_var[:].squeeze(),
+        {"long_name": lon_var.long_name, "units": lon_var.units},
+    )
+    ds["lat"] = (
+        ("x", "y"),
+        lat_var[:].squeeze(),
+        {"long_name": lat_var.long_name, "units": lat_var.units},
+    )
+    ds["time"] = (
+        ("time",),
+        time_var[:].squeeze(),
+        {"long_name": time_var.long_name, "units": time_var.units},
+    )
+    ds["lon"] = ds["lon"].fillna(-999.99)
+    ds["lat"] = ds["lat"].fillna(-999.99)
+    ds = ds.set_coords(["time", "lon", "lat"])
+
+    ds.attrs["reference_time_string"] = dso.time_coverage_start
+    ds.attrs["granule_number"] = dso.granule_num
+    ds.attrs["scan_num"] = dso.scan_num
+
+    for varname in variable_dict:
+        values_var = dso.groups["product"][varname]
+        values = values_var[:].squeeze()
+        fv = values_var.getncattr("_FillValue")
+        if not np.isfinite(fv):
```

Comment:
What is the rationale behind this condition? `_FillValue` could be set to anything.

Reply:
For example, the default nc fill value for double, even though it's large, is still finite, so the replacement by NaN will be skipped with this current `if not np.isfinite(fv):` guard.
```python
>>> np.isfinite(netCDF4.default_fillvals["f8"])
True
```

Reply:
Done.